### PR TITLE
[CodeComplete] Skip conjunction elements unrelated to code completion token 🚥 #59944

### PR DESCRIPTION
**Pull request by ahoppen**
Depends on https://github.com/apple/swift/pull/59944

---

This should improve performance of code completion inside result builders by not type checking conjunction elements that don’t affect code completion.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/723

@swift-ci Please smoke test macOS

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

File: lib/Sema/CSSyntacticElement.cpp [hunk 1]
```diff
@@ -995,6 +995,10 @@ class SyntacticElementConstraintGenerator
         }
       }
 
+      if (cs.isForCodeCompletion() && !cs.containsIDEInspectionTarget(element)) {
```
Comment (xedin) [inline lib/Sema/CSSyntacticElement.cpp:998]: Nit: could you please a comment here with brief explanation why this is necessary?

**Review by xedin — APPROVED** [review]
LGTM! I think this is worth a cherry-pick to 5.8

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
@swift-ci Please SourceKit stress test

**Comment by ahoppen** [comment]
This is now part of https://github.com/apple/swift/pull/63717

**Pull request closed without merging**